### PR TITLE
removed code that deal with potential null nodes (not possible anymore)

### PR DESCRIPTION
**Pull request by fabpot**

### CODE REVIEW

File: lib/Twig/NodeTraverser.php [hunk 1]
```diff
@@ -70,10 +70,6 @@ public function traverse(Twig_Node $node)
 
     private function traverseForVisitor(Twig_NodeVisitorInterface $visitor, Twig_Node $node = null)
     {
-        if (null === $node) {
```
Comment (stof) [inline lib/Twig/NodeTraverser.php]: you forgot to change the method signature

File: test/Twig/Tests/NodeVisitor/OptimizerTest.php [hunk 2]
```diff
@@ -89,10 +89,6 @@ public function getTestsForForOptimizer()
 
     public function checkForConfiguration(Twig_Node $node = null, $target, $withLoop)
     {
-        if (null === $node) {
```
Comment (stof) [inline test/Twig/Tests/NodeVisitor/OptimizerTest.php]: same here about the method signature

File: lib/Twig/NodeTraverser.php (hunk 1 above)
Comment (fabpot) [inline lib/Twig/NodeTraverser.php]: fixed

File: test/Twig/Tests/NodeVisitor/OptimizerTest.php (hunk 2 above)
Comment (fabpot) [inline test/Twig/Tests/NodeVisitor/OptimizerTest.php]: done

**Pull request merged**